### PR TITLE
Add exception chaining to self key encryption

### PR DESCRIPTION
**Pull request by sitaram-kalluri**
**- What I did**
1. Add exception chain when SelfEncryptionKey not found.
2. Add unit test for positive and negative scenario of Self Key Encryption.
3. Add NullCheck extension on String to check if String is null or empty.

**- How I did it**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: at_client/lib/src/encryption_service/self_key_encryption.dart [hunk 1]
```diff
@@ -9,35 +11,45 @@ import 'package:at_utils/at_logger.dart';
...
     try {
-      // Get AES key for current atSign
-      var selfEncryptionKey = await _getSelfEncryptionKey();
-      if (selfEncryptionKey == null ||
-          selfEncryptionKey.isEmpty ||
-          selfEncryptionKey == 'data:null') {
-        throw KeyNotFoundException(
-            'Self encryption key is not set for atSign ${atKey.sharedBy}');
+      selfEncryptionKey = await localSecondary.getEncryptionSelfKey();
+      if (selfEncryptionKey.isNull || selfEncryptionKey == 'data:null') {
```
Comment (murali-shris) [inline at_client/lib/src/encryption_service/self_key_encryption.dart]: local secondary getEncryptionSelfKey has a call to keystore.get..keystore.get will not return data:null.. do we need the check?

Comment (sitaram-kalluri) [reply at_client/lib/src/encryption_service/self_key_encryption.dart]: Yeah, the 'data:null' check can be ignored. I will remove the condition and push the changes.

**Review by murali-shris — APPROVED** [review]
